policy/pa.go

```diff
@@ -174,26 +178,32 @@ func isDNSCharacter(ch byte) bool {
-	errEmptyName            = berrors.MalformedError("DNS name was empty")
-	errNameEndsInDot        = berrors.MalformedError("DNS name ends in a period")
-	errTooFewLabels         = berrors.MalformedError("DNS name does not have enough labels")
-	errLabelTooShort        = berrors.MalformedError("DNS label is too short")
-	errLabelTooLong         = berrors.MalformedError("DNS label is too long")
-	errMalformedIDN         = berrors.MalformedError("DNS label contains malformed punycode")
-	errInvalidRLDH          = berrors.RejectedIdentifierError("DNS name contains a R-LDH label")
-	errTooManyWildcards     = berrors.MalformedError("DNS name had more than one wildcard")
-	errMalformedWildcard    = berrors.MalformedError("DNS name had a malformed wildcard label")
-	errICANNTLDWildcard     = berrors.MalformedError("DNS name was a wildcard for an ICANN TLD")
-	errWildcardNotSupported = berrors.MalformedError("Wildcard names not supported")
+	errNonPublic            = berrors.MalformedError("Domain name does not end with a valid public suffix (TLD)")
+	errICANNTLD             = berrors.MalformedError("Domain name is an ICANN TLD")
+	errPolicyForbidden      = berrors.RejectedIdentifierError("The ACME server refuses to issue a certificate for this domain name, because it is forbidden by policy")
+	errInvalidDNSCharacter  = berrors.MalformedError("Domain name contains an invalid character")
+	errNameTooLong          = berrors.MalformedError("Domain name is longer than 230 bytes, which is the maximum length supported by this server")
+	errIPAddress            = berrors.MalformedError("The ACME server can not issue a certificate for an IP address")
+	errTooManyLabels        = berrors.MalformedError("Domain name has more than 10 labels (parts)")
+	errEmptyName            = berrors.MalformedError("Domain name is empty")
+	errNameEndsInDot        = berrors.MalformedError("Domain name ends in a dot")
+	errTooFewLabels         = berrors.MalformedError("Domain name needs at least one dot")
+	errLabelTooShort        = berrors.MalformedError("Domain name can not have two dots in a row")
+	errLabelTooLong         = berrors.MalformedError("Domain has a label (component between dots) longer than 63 bytes")
+	errMalformedIDN         = berrors.MalformedError("Domain name contains malformed punycode")
+	errInvalidRLDH          = berrors.RejectedIdentifierError("Domain name contains an invalid label in a reserved format (R-LDH: '??--')")
+	errTooManyWildcards     = berrors.MalformedError("Domain name has more than one wildcard")
+	errMalformedWildcard    = berrors.MalformedError("Domain name contains an invalid wildcard. A wildcard is only permitted before the first dot in a domain name")
+	errICANNTLDWildcard     = berrors.MalformedError("Domain name is a wildcard for an ICANN TLD")
+	errWildcardNotSupported = berrors.MalformedError("Wildcard domain names are not supported")
 )
 
+// In these error messages:
```

Comment:
I think this comment should also be before the variables in question, not after. Perhaps on L178 before the `var (`?

Reply:
moved